clients/src/main/resources/common/message/ProduceRequest.json

```diff
@@ -33,21 +33,21 @@
   "validVersions": "0-8",
   "flexibleVersions": "none",
   "fields": [
-    { "name": "TransactionalId", "type": "string", "versions": "3+", "nullableVersions": "0+", "entityType": "transactionalId",
+    { "name": "TransactionalId", "type": "string", "versions": "3+", "nullableVersions": "3+", "ignorable": true, "entityType": "transactionalId",
       "about": "The transactional ID, or null if the producer is not transactional." },
     { "name": "Acks", "type": "int16", "versions": "0+",
       "about": "The number of acknowledgments the producer requires the leader to have received before considering a request complete. Allowed values: 0 for no acknowledgments, 1 for only the leader and -1 for the full ISR." },
-    { "name": "TimeoutMs", "type": "int32", "versions": "0+",
+    { "name": "Timeout", "type": "int32", "versions": "0+",
```

Comment:
nit: I kind of liked the original name to make the unit clear. We're probably not consistent on this convention though.

Reply:
This change is for compatibility. The name used by origin serialization is "timeout" rather than "timeoutMs" (see https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java#L54).

Reply:
The names do not get serialized, so I think we can make them whatever we want.

Reply:
This is a major question when I am processing those protocol migration PRs.

The Struct deserialization depends on the "name" (https://github.com/apache/kafka/blob/8e211eb72f9a45897cc37fed394a38096aa47feb/clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java#L250). Struct.get(xxx) searches the index according to "name". Hence, changing the name breaks compatibility since the Struct serialization can not find the correct field. (That is to say, the previous protocol fails to deserialize the data from new auto-generated protocol)

Please correct me if I misunderstood protocol mechanism.

Reply:
oh, I trace the code again. You are right. The name is not serialized.

I will revise the field names as you suggested.